docs/developer-docs/smart-contracts/signatures/signing-messages-tschnorr.mdx

```diff
@@ -0,0 +1,308 @@
+
+To deploy code using the `sign_with_schnorr` method, you will need to specify the correct `key_ID` for the environment in the canister's source code. The following key IDs are supported:
+
+- `dfx_test_key1`: Only available on the local replica started by dfx.
+- `test_key_1`: Test key available on the ICP mainnet.
+- `key_1`: Production key available on the ICP mainnet.
+
+<AdornedTabs groupId="language">
+<TabItem value="motoko" label="Motoko" default>
+
+```motoko
+  public shared ({ caller }) func sign(message_arg : Text, algorithm_arg : SchnorrAlgotirhm) : async {
+    #Ok : { signature_hex : Text };
+    #Err : Text;
+  } {
+    try {
+      Cycles.add(25_000_000_000);
+      let { signature } = await ic.sign_with_schnorr({
+        message = Text.encodeUtf8(message_arg);
+        derivation_path = [Principal.toBlob(caller)];
+        key_id = { algorithm = algorithm_arg; name = "dfx_test_key" };
+      });
+      #Ok({ signature_hex = Hex.encode(Blob.toArray(signature)) });
+    } catch (err) {
+      #Err(Error.message(err));
+    };
+  };
+```
+
+</TabItem>
+
+<TabItem value="rust" label="rust">
+
+```rust
+#[update]
+async fn sign(message: String, algorithm: SchnorrAlgorithm) -> Result<SignatureReply, String> {
+    let internal_request = ManagementCanisterSignatureRequest {
+        message: message.as_bytes().to_vec(),
+        derivation_path: vec![ic_cdk::api::caller().as_slice().to_vec()],
+        key_id: SchnorrKeyIds::TestKeyLocalDevelopment.to_key_id(algorithm),
+    };
+
+    let (internal_reply,): (ManagementCanisterSignatureReply,) =
+        ic_cdk::api::call::call_with_payment(
+            mgmt_canister_id(),
+            "sign_with_schnorr",
+            (internal_request,),
+            25_000_000_000,
+        )
+        .await
+        .map_err(|e| format!("sign_with_schnorr failed {e:?}"))?;
+
+    Ok(SignatureReply {
+        signature_hex: hex::encode(&internal_reply.signature),
+    })
+}
+```
+
+</TabItem>
+</AdornedTabs>
+
+### Cost
+
+The `sign_with_schnorr` method requires cycles to be attached. When the method in the Rust CDK is used, the required cycles are automatically attached. In Motoko, the cycles need to be added manually.
+
+[Learn more about threshold Schnorr costs](/docs/current/developer-docs/gas-cost).
+
+## Verifying signatures
+
+Created signatures can be verified with the public key corresponding to the canister that initiated the request and the derivation path. For threshold Schnorr, Ed25519 and BIP340 are verified differently.
+
+Calls to the `verify` method should always return `true` if the correct parameters are provided or return `false` otherwise.
+
+BIP340 can be verified using code such as:
+
+:::caution
+The first byte of the BIP340 public key needs to be removed for verification. Below, this is done by the verification function internally.
+:::
+
+<AdornedTabs groupId="language">
+<TabItem value="motoko" label="Motoko" default>
+
+```motoko
```

Comment:
this is actually JS I believe because there are no libraries in motoko for signature verification because of the implementation complexity, i.e., we are doing the verification in the frontend, not in the canister (in contrast to the rust example; confusing, I agree, but I think it's still good to show different ways to verify signatures because there will be different use cases)

Reply:
Fixed language